internal/grpctest/tlogger/tlogger.go

```diff
@@ -0,0 +1,102 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package tlogger initializes the testing logger on import which logs to the
+// testing package's T struct.
+package tlogger
+
+import (
+	"os"
+	"strconv"
+	"testing"
+
+	"google.golang.org/grpc/grpclog"
+)
+
+var logger = tLogger{v: 0}
+
+type tLogger struct {
+	v int
+	t *testing.T
+}
+
+func init() {
+	vLevel := os.Getenv("GRPC_GO_LOG_VERBOSITY_LEVEL")
```

Comment:
I actually think it should be OK to log everything (all verbose levels) in this test logger.

`t.Log` only logs when the test failed, so we won't get anything for the passed tests. For the failed test, the v2 logs could be helpful.

Reply:
I agree V2 should be fine by default.  I don't think we should set this to 99 by default, though.  E.g. we might add a V10 that logs every byte written to the wire, and that would be too much to do by default, but could still be useful when running a single small test case.